embed/serve_test.go

```diff
@@ -25,14 +27,48 @@ import (
 // TestStartEtcdWrongToken ensures that StartEtcd with wrong configs returns with error.
 func TestStartEtcdWrongToken(t *testing.T) {
 	tdir, err := ioutil.TempDir(os.TempDir(), "token-test")
+
 	if err != nil {
 		t.Fatal(err)
 	}
 	defer os.RemoveAll(tdir)
+
 	cfg := NewConfig()
+
+	// Same functions as used in integration/embed/embed_test.go for setting up Config.
+	urls := newEmbedURLs(false, 2)
+	setupEmbedCfg(cfg, []url.URL{urls[0]}, []url.URL{urls[1]})
+
 	cfg.Dir = tdir
 	cfg.AuthToken = "wrong-token"
+
 	if _, err = StartEtcd(cfg); err != auth.ErrInvalidAuthOpts {
 		t.Fatalf("expected %v, got %v", auth.ErrInvalidAuthOpts, err)
 	}
 }
+
+func newEmbedURLs(secure bool, n int) (urls []url.URL) {
+	scheme := "unix"
+	if secure {
+		scheme = "unixs"
+	}
+	for i := 0; i < n; i++ {
+		u, _ := url.Parse(fmt.Sprintf("%s://localhost:%d%06d", scheme, os.Getpid(), i))
+		urls = append(urls, *u)
+	}
+	return urls
+}
+
+func setupEmbedCfg(cfg *Config, curls []url.URL, purls []url.URL) {
```

Comment:
Sorry I do not follow why do we need this function. It looks like many settings are the default values in NewConfig()?

Reply:
Some of the settings here can be removed. This was copied (along with the other function) from [integration/embed/embed_test.go](https://github.com/etcd-io/etcd/blob/master/integration/embed/embed_test.go). I can move the logic fromm this function back into the main test function and clean up the settings to the minimal required if that would be cleaner.

Reply:
I made the updated to make things cleaner and squashed the commits again, please let me know if update resolves this issue.